src/MOIWrapper.jl

```diff
@@ -0,0 +1,397 @@
+export CbcOptimizer, copy!, optimize!, get
+
+
+using MathOptInterface
+using Cbc.CbcCInterface
+
+const MOI = MathOptInterface
+const MOIU = MathOptInterface.Utilities
+const CbcCI = CbcCInterface
+
+const SUPPORTED_OBJECTIVES = [
+    MOI.ScalarAffineFunction{Float64}
+]
+
+const SUPPORTED_CONSTRAINTS = [
+    (MOI.ScalarAffineFunction{Float64}, MOI.Interval{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.EqualTo{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.GreaterThan{Float64}),
+    (MOI.SingleVariable, MOI.EqualTo{Float64}),
+    (MOI.SingleVariable, MOI.LessThan{Float64}),
+    (MOI.SingleVariable, MOI.GreaterThan{Float64}),
+    (MOI.SingleVariable, MOI.Interval{Float64}),
+    (MOI.SingleVariable, MOI.ZeroOne),
+    (MOI.SingleVariable, MOI.Integer)
+]
+
+const BOUND_CONSTRAINTS = [
+    (MOI.ScalarAffineFunction{Float64}, MOI.Interval{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.EqualTo{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.GreaterThan{Float64}),
+    (MOI.SingleVariable, MOI.EqualTo{Float64}),
+    (MOI.SingleVariable, MOI.LessThan{Float64}),
+    (MOI.SingleVariable, MOI.GreaterThan{Float64}),
+    (MOI.SingleVariable, MOI.Interval{Float64})
+]
+
+const INTEGRALITY_CONSTRAINTS = [
+    (MOI.SingleVariable, MOI.ZeroOne),
+    (MOI.SingleVariable, MOI.Integer)
+]
+
+mutable struct CbcOptimizer <: MOI.AbstractOptimizer
+    inner::CbcModel
+    # env
+    objFunctConstant::Float64
+    CbcOptimizer() = new(CbcModel(), 0.0) # Initializes with an empty model
+end
+
+mutable struct CbcModelFormat
+    nbRows::Int
+    nbCols::Int
+    constraint_matrix::Array{Float64,2}
```

Comment:
You probably want to make this a sparse array instead of a dense array. See `? SparseMatrixCSC`